File: UVA/10945_Problem C: Mother Bear/solve.cc
#include <cctype>
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

std::vector<char> punctuation_and_white_space = std::vector<char>{'.', ',', '!', '?', ' '};
#define IN_CONTAINER(elem, container) (std::find((container.begin()), (container).end(), (elem)) != (container).end())
#define NOT_IN_CONTAINER(elem, container) !(IN_CONTAINER((elem), (container)))

std::string strip(const std::string & s) {
    std::string res;
    for (const auto & c : s) {
         bool skip = false;
         if (NOT_IN_CONTAINER(c, punctuation_and_white_space)) {
             res += c;
         }
    }
    std::transform(res.begin(), res.end(), res.begin(), ::tolower);
//    std::cout << res << std::endl;
    return res;
}

bool is_palindromes(const std::string & s) {
    for (int i = 0, k = s.size()-1; i < k; ++i, --k) {
        if (s[i] != s[k])
            return false;
    }
    return true;
}

int main() {

    std::string line;
    while (std::getline(std::cin, line) && line != "DONE") {
        if (is_palindromes(strip(line))) {
            std::cout << "You won't be eaten!" << std::endl;
        } else {
            std::cout << "Uh oh.." << std::endl;
        }
    }

    return 0;
}

File: UVA/10945_Problem C: Mother Bear/solve.py
import string;
def is_palindrome(stmt):
    stmt = [c for c in stmt if c not in string.punctuation and c not in string.whitespace];
    stmt = ''.join(stmt).lower();
#    print(stmt);
    i, k = 0, len(stmt)-1;
    while True:
        if i >= k:
            break;
        if stmt[i] != stmt[k]:
            return False;
        i, k = i+1, k-1;
    return True;
    # print(stmt);

try:
    while True:
        line = input();
        if line == "DONE":
            break;
        if is_palindrome(line):
            print("You won't be eaten!");
        else: 
            print("Uh oh..");
except EOFError:
    pass;
